packages/@lwc/shared/src/aria.ts

```diff
@@ -74,21 +74,36 @@ const AriaPropertyNames = [
     'role',
 ] as const;
 
+type AriaProperty = (typeof AriaPropertyNames)[number];
+
 export type AccessibleElementProperties = {
-    [prop in (typeof AriaPropertyNames)[number]]: string | null;
+    [Prop in AriaProperty]: string | null;
+};
+
+type AriaPropToAttrMap = {
+    [Prop in AriaProperty]: Prop extends `aria${infer S}` ? `aria-${Lowercase<S>}` : Prop;
```

Comment:
This may break someday if we decide to ever support `ariaLabelledByElements`/`ariaActiveDescendantElement` and friends, but we can cross that bridge when we get there.